File: linked-list/remove-dups-java/src/test/java/SolutionV1Test.java
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class SolutionV1Test {

    @Test
    public void removeDupsTest() {
        SolutionV1 solutionV1 = new SolutionV1();

        Node node4 = new Node(null, 1);
        Node node3 = new Node(node4, 3);
        Node node2 = new Node(node3, 2);
        Node node1 = new Node(node2, 1);

        solutionV1.removeDups(node1);
        List<Integer> expected = new ArrayList<>();
        extracted(node1, expected);
        Assertions.assertEquals(expected, List.of(new Integer[]{1, 2, 3}));
    }

    private static void extracted(Node node, List<Integer> expected) {
        expected.add(node.getData());

        if (node.getNext() == null) {
            return;
        }
        extracted(node.getNext(), expected);
    }
}

File: tree-and-graphs/route-between-nodes-go/solution_v1_test.go
package route_between_nodes_go

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestSolutionV1(t *testing.T) {
	node1 := Node{
		Value: 1,
	}

	node2 := Node{
		Value: 2,
	}

	node3 := Node{
		Value: 3,
	}

	node4 := Node{
		Value: 4,
	}

	node5 := Node{
		Value: 5,
	}
	node3.Adjacents = []Node{
		node5,
	}

	node1.Adjacents = []Node{
		node2,
		node3,
	}

	assert.True(t, SolutionV1(node1, node2))
	assert.True(t, SolutionV1(node1, node5))
	assert.False(t, SolutionV1(node1, node4))
}
